clang/lib/CIR/CodeGen/CIRGenCXXABI.cpp

```diff
@@ -85,3 +85,86 @@ bool CIRGenCXXABI::isZeroInitializable(const MemberPointerType *MPT) {
   // Fake answer.
   return true;
 }
+
+void CIRGenCXXABI::errorUnsupportedABI(CIRGenFunction &CGF, StringRef S) {
+  DiagnosticsEngine &Diags = CGF.CGM.getDiags();
+  unsigned DiagID = Diags.getCustomDiagID(DiagnosticsEngine::Error,
+                                          "cannot yet compile %0 in this ABI");
+  Diags.Report(CGF.getContext().getFullLoc(CGF.CurCodeDecl->getLocation()),
+               DiagID)
+      << S;
+}
+
+CharUnits CIRGenCXXABI::getArrayCookieSize(const CXXNewExpr *E) {
+  if (!requiresArrayCookie(E))
+    return CharUnits::Zero();
+  return getArrayCookieSizeImpl(E->getAllocatedType());
+}
+
+CharUnits CIRGenCXXABI::getArrayCookieSizeImpl(QualType ElementType) {
+  // BOGUS
```

Comment:
why? should an unreachable be placed here for now?

Reply:
I copied this directly from the equivalent code in the non-CIR codegen. including the comment. An unreachable is probably appropriate or maybe even making this function pure virtual. The expectation is that any subclass will override this function. I think the comment is correct in as much as this will never be the correct behavior if we get here.

Reply:
Oh boy! Thanks for the clarification